osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -284,7 +289,17 @@ public override bool OnExiting(ScreenExitEvent e)
                 if (dialogOverlay.CurrentDialog is ConfirmExitDialog exitDialog)
                     exitDialog.PerformOkAction();
                 else
-                    dialogOverlay.Push(new ConfirmExitDialog(confirmAndExit, () => exitConfirmOverlay.Abort()));
+                {
+                    dialogOverlay.Push(new ConfirmExitDialog(() =>
+                    {
+                        InvokeWhenExiting?.Invoke();
+                        confirmAndExit();
+                    }, () =>
+                    {
+                        exitConfirmOverlay.Abort();
+                        InvokeWhenExiting = null;
```

Comment:
I am not personally feeling this solution... `InvokeWhenExiting` is a public field, that needs to be set with the correct ordering to work properly, and gets unset after the next exit dialog is dismissed. It's maybe not as bad as killing processes, but very indirect as far as data flow goes, and still inside "I don't like it" territory as far as I'm concerned.

I don't have anything to offer as the proper solution but at least I can give this: the criteria for me to like the final solution is to figure out a clean flow that passes the delegate untouched as long as it needs to until the point of no return where it *is* safe to call squirrel.

Reply:
The criteria would also have to take into account whether the exit was aborted during the exit flow process, since at that point the user would expect the game not to restart when they manually exit. Can see why this is hard to get working now...